OpenSim/Common/Component.cpp

```diff
@@ -708,6 +649,8 @@ const Component::StateVariable* Component::
     // top level component
     std::string::size_type back = name.rfind("/");
     std::string prefix = name.substr(0, back);
+    // this is a bug (or misuse?)! if the name does not have a "/", then
+    // name = prefix = varName. back+1 = 0 due to uint
```

Comment:
This was intentional (mis)use so that `varName` is `name` if there are no `/` in the `name`. The effect when `back = string::max_size` (not found) is that of `name.length() - back = name.length() +1 `. I stumbled upon on that fact and left it.